### PR TITLE
e2e groupby conditional scenarios

### PR DESCRIPTION
**Pull request by itsmekumari**
Design document available in PR review folder.

### CODE REVIEW

File: core-plugins/src/e2e-test/features/groupby/GroupByWithFile.feature [hunk 1]
```diff
@@ -153,3 +153,156 @@ Feature: GroupBy - Verify File source to File sink data transfer using GroupBy a
...
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
+    Then Validate OUT record count of groupby is equal to IN record count of sink
+    Then Validate output file generated by file sink plugin "fileSinkTargetBucket" is equal to expected output file "groupByTest5OutputFile"
+
+  @GROUP_BY_TEST @FILE_SINK_TEST
+  Scenario: To verify complete flow of data extract and transfer from File source to File sink with GroupBy plugin using set of aggregates
```
Comment (vanathi-g) [inline core-plugins/src/e2e-test/features/groupby/GroupByWithFile.feature]: nit: update scenario description to mention which aggregates are being tested (count, sum, max)

**Review by vanathi-g — COMMENTED** [review]
Also, just a general question, why were minIf and anyIf kept in a separate scenario? Why not keep them in the first scenario itself?

File: core-plugins/src/e2e-test/features/groupby/GroupByWithFile.feature (hunk 1 above)
Comment (itsmekumari) [reply core-plugins/src/e2e-test/features/groupby/GroupByWithFile.feature]: Updated scenario description.

**Comment by itsmekumari** [comment]
> Also, just a general question, why were minIf and anyIf kept in a separate scenario? Why not keep them in the first scenario itself?

Added minif and anyif aggregators in first scenario.

**Review by vanathi-g — APPROVED** [review]
Seems like the build failure might be flaky behaviour, please validate that once. LGTM otherwise. Ensure e2e test build works before merging

**Comment by vanathi-g** [comment]
e2e test failure seems to be due to "Verify Distinct plugin validation errors for incorrect data in Fields" (src/e2e-test/features/distinctplugin/DistinctErrorScenarios.feature:5) scenario. Merging since it is unrelated.

**Pull request merged**